packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -507,17 +510,41 @@ function voidFunctionParams(
       : subType.getConstructSignatures();
     for (const signature of signatures) {
       for (const [index, parameter] of signature.parameters.entries()) {
-        const type = checker.getTypeOfSymbolAtLocation(
+        const decl = parameter.getDeclarations()?.[0];
```

Comment:
This is a bit more direct. I also got a little confused reading this about why we'd consider there to be more than one declaration.

```suggestion
        const decl = parameter.valueDeclaration;
```

Reply:
I've adjusted this to use `valueDeclaration`